### PR TITLE
build(stochastic-test-utils): use node16 module resolution

### PR DESCRIPTION
**Pull request by DLehenbauer**
Semi-automatic PR to switch to node16 module resolution

### CODE REVIEW

File: packages/dds/test-dds-utils/src/test/ddsFuzzHarness.spec.ts [hunk 1]
```diff
@@ -2,15 +2,18 @@
  * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
  * Licensed under the MIT License.
  */
-
+/* eslint-disable @typescript-eslint/no-unsafe-return */
+/* eslint-disable @typescript-eslint/no-unsafe-call */
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
```
Comment (tylerbutler) [inline packages/dds/test-dds-utils/src/test/ddsFuzzHarness.spec.ts:8]: This seems suspicious... it implies that typescript wasn't finding types for some of the APIs used here.

File: packages/dds/test-dds-utils/src/test/ddsFuzzHarness.spec.ts [hunk 2]
```diff
@@ -2,15 +2,18 @@
...
 import { strict as assert } from "node:assert";
 import * as fs from "node:fs";
 import * as path from "node:path";
 
 import { TypedEventEmitter } from "@fluid-internal/client-utils";
 import type { AsyncGenerator } from "@fluid-private/stochastic-test-utils";
 import { chainAsync, done, takeAsync } from "@fluid-private/stochastic-test-utils";
-// eslint-disable-next-line import/no-internal-modules
+// eslint-disable-next-line import/no-unresolved
```
Comment (tylerbutler) [inline packages/dds/test-dds-utils/src/test/ddsFuzzHarness.spec.ts:16]: Do we understand why this can't be resolved?

**Review by tylerbutler — CHANGES_REQUESTED** [review]
Some lint disables seem suspicious...

File: packages/dds/test-dds-utils/src/test/ddsFuzzHarness.spec.ts (hunk 2 above)
Comment (sonalideshpandemsft) [reply packages/dds/test-dds-utils/src/test/ddsFuzzHarness.spec.ts:13]: The `./test` exports were incorrect. fixed this

File: packages/dds/test-dds-utils/src/test/ddsFuzzHarness.spec.ts (hunk 1 above)
Comment (sonalideshpandemsft) [reply packages/dds/test-dds-utils/src/test/ddsFuzzHarness.spec.ts]: It appears that I had misconfigured the exports for the test files.

I noticed that the utils.d.ts file wasn't being published to either the lib or dist folder. After adding `declaration: true` to the tsconfig.json eslint errors were resolved locally.

**Review by tylerbutler — APPROVED** [review]
Looks good now that the exports are correct.

**Pull request merged**